### DISCUSSION TITLE
TensorFlow Question

### DISCUSSION THREAD

**Discussion by sohail579** [immich-app/immich#584] [Q&A]
I am a big fan of this project its about time someone put something like this together.   so bid thanks to the Dev.

I have a question about TensorFlow.  I have a Google Coral will this work?  I use it for my CCTV but would imagine that it should work here too?

**Comment by alextran1502** [comment]
I haven't used Google Coral, let me take a look at is and see if we can make a connector for that system

**Reply by DrMxrcy** [reply]
It would be a great feature to have. Would allow machine learning on lower powered devices

**Comment by xe2io** [comment]
I think the [accelerators](https://coral.ai/products/) only support TensorFlow Lite, so maybe the model just needs to be converted?  (I'm very new to ML so am not familiar with the terminology/how it all works.)

I'm also interested in this- finally received a m.2 accelerator (order placed in Nov '21) so will be exploring this further, but need to first deploy immich and install the card itself.  Hopefully I'll have some interesting info soon :)

**Reply by sohail579** [reply]
i have have the USB version.  will hold out until the Unraid image is released though

**Reply by DrMxrcy** [reply]
@alextran1502 Any Update on Google Coral? Frigate has integration for it as an example! https://docs.frigate.video/

**Reply by alextran1502** [reply]
@DrMxrcy i don't have any hardware for testing and developing on Google Coral at the moment.

**Reply by pyro2927** [reply]
@alextran1502 I realize this is probably not high on the priority list, but I'd be happy to send a Coral device your way if that helps.

**Comment by tristan-k** [comment]
Any update on coral support (m.2 or usb) for immich?

**Comment by bo0tzz** [comment]
We looked into coral at some point but my understanding is they're not powerful enough to run the sort of models we use. We do have a more modern accelerator on the shelf to take a look at when time permits.
